stubs/Flask-SQLAlchemy/flask_sqlalchemy/model.pyi

```diff
@@ -10,14 +14,14 @@ camelcase_re: Pattern[str]
 def camel_to_snake_case(name: str) -> str: ...
 
 class NameMetaMixin(type):
-    def __init__(cls, name, bases, d) -> None: ...
-    def __table_cls__(cls, *args, **kwargs): ...
+    def __init__(cls, name: str, bases: tuple[type, ...], d: dict[str, Any]) -> None: ...
+    def __table_cls__(cls, *args: Any, **kwargs: Any) -> Table | None: ...
 
 class BindMetaMixin(type):
-    def __init__(cls, name, bases, d) -> None: ...
+    def __init__(cls, name: str, bases: tuple[type, ...], d: dict[str, Any]) -> None: ...
 
 class DefaultMeta(NameMetaMixin, BindMetaMixin, DeclarativeMeta): ...
 
-class Model:
-    query_class: Any | None
-    query: Any | None
+class Model(Generic[_Model]):
+    query_class: type[Query[_Model]] | None
+    query: Query[_Model] | None
```

Comment:
@srittau, does making `Model` generic look correct to you?

Reply:
Class implementation can be seen here, if it helps:
https://github.com/pallets-eco/flask-sqlalchemy/blob/main/src/flask_sqlalchemy/model.py#L115

Reply:
I have no experience with flask-sqlalchemy, but considering that `sqlalchemy.orm.query` is generic: probably.

Reply:
Let's YOLO it then. If it causes a problem for somebody, we'll know soon enough.